### PR TITLE
make AS a trait to get a clean generic param

### PR DESCRIPTION
**Pull request by smtmfft**

### CODE REVIEW

File: snark-verifier-sdk/src/lib.rs [hunk 1]
```diff
@@ -120,9 +124,9 @@ pub trait CircuitExt<F: Field>: Circuit<F> {
 }
 
 pub fn read_pk<C: CurveAffine<ScalarExt = Fr>, ConcreteCircuit: Circuit<C::Scalar>>(
-        path: &Path,
-        #[cfg(feature = "halo2_circuit_params")] params: ConcreteCircuit::Params,
-    ) -> io::Result<ProvingKey<G1Affine>> {
+    path: &Path,
+    #[cfg(feature = "halo2_circuit_params")] params: ConcreteCircuit::Params,
+) -> io::Result<ProvingKey<G1Affine>> {
```
Comment (Brechtpd) [inline snark-verifier-sdk/src/lib.rs]: Only formatting change

File: snark-verifier-sdk/src/lib.rs [hunk 2]
```diff
@@ -137,8 +141,10 @@ pub fn read_pk<C: CurveAffine<ScalarExt = Fr>, ConcreteCircuit: Circuit<C::Scala
     let pk = ProvingKey::read::<_, ConcreteCircuit>(
         &mut bufreader,
         SerdeFormat::RawBytes,
-        #[cfg(feature = "halo2_circuit_params")] params,
-    ).unwrap();
+        #[cfg(feature = "halo2_circuit_params")]
+        params,
+    )
+    .unwrap();
```
Comment (Brechtpd) [inline snark-verifier-sdk/src/lib.rs]: Only formatting change

File: snark-verifier-sdk/src/lib.rs [hunk 3]
```diff
@@ -156,7 +162,8 @@ pub fn gen_pk<C: CurveAffine<ScalarExt = Fr>, ConcreteCircuit: Circuit<C::Scalar
     if let Some(path) = path {
         if let Ok(pk) = read_pk::<C, ConcreteCircuit>(
             path,
-            #[cfg(feature = "halo2_circuit_params")] params2,
+            #[cfg(feature = "halo2_circuit_params")]
+            params2,
```
Comment (Brechtpd) [inline snark-verifier-sdk/src/lib.rs]: Only formatting change

**Review by Brechtpd — APPROVED** [review]
Only feedback is that there are some formatting only changes that probably should be reverted, but otherwise looks good to me!

**Comment by smtmfft** [comment]
The format changes are reverted, please help to merge, then I will update circuit & chain.

**Comment by smtmfft** [comment]
Submitted same PR to PSE, see if it will be accepted.
https://github.com/privacy-scaling-explorations/snark-verifier/pull/43

**Pull request merged**